test/tests/functional/pbs_cray_vnode_pool.py

```diff
@@ -0,0 +1,325 @@
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+@tags('cray', 'configuration')
+class TestVnodePool(TestFunctional):
+
+    """
+    This test suite tests how PBS makes use of node attribute "vnode_pool"
+    It expects at least 2 moms to be specified to it while executing.
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+
+        # The moms provided to the test may have unwanted vnodedef files.
+        if self.moms.values()[0].has_vnode_defs():
+            self.moms.values()[0].delete_vnode_defs()
+        if self.moms.values()[1].has_vnode_defs():
+            self.moms.values()[1].delete_vnode_defs()
+
+        # Check if vnodes exist before deleting nodes.
+        # Clean all default nodes because each test case will set up nodes.
+        try:
+            self.server.status(NODE)
+            rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+            self.assertEqual(rc, 0)
+        except PbsStatusError as e:
+            self.assertEquals(
+                e.msg[0], "/opt/pbs/bin/pbsnodes: Server has no node list")
+
+    def test_invalid_values(self):
+        """
+        Invalid vnode_pool values shall result in errors.
+        """
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        attr_A = {'vnode_pool': '-1'}
+        try:
+            self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA,
+                                attrib=attr_A, expect=False)
+        except PbsManagerError as e:
+            self.assertEquals(
+                e.msg[1], "qmgr: Error (15014) returned from server")
+
+        attr_A = {'vnode_pool': '0'}
```

Comment:
Just curious, why is this called attr_A when you use just attr down below?  It confused me a little.